crates/subspace-farmer/src/single_disk_farm/plot_cache.rs

```diff
@@ -0,0 +1,265 @@
+#[cfg(test)]
+mod tests;
+
+use async_lock::RwLock as AsyncRwLock;
+use parking_lot::RwLock;
+use std::collections::HashMap;
+use std::fs::File;
+use std::sync::{Arc, Weak};
+use std::{io, mem};
+use subspace_core_primitives::crypto::blake3_hash_list;
+use subspace_core_primitives::{Blake3Hash, Piece, PieceIndex, SectorIndex};
+use subspace_farmer_components::file_ext::FileExt;
+use subspace_farmer_components::sector::SectorMetadataChecksummed;
+use subspace_networking::libp2p::kad::RecordKey;
+use subspace_networking::utils::multihash::ToMultihash;
+use thiserror::Error;
+use tracing::{debug, info, warn};
+
+/// Disk plot cache open error
+#[derive(Debug, Error)]
+pub enum DiskPlotCacheError {
+    /// I/O error occurred
+    #[error("I/O error: {0}")]
+    Io(#[from] io::Error),
+    /// Checksum mismatch
+    #[error("Checksum mismatch")]
+    ChecksumMismatch,
+}
+
+#[derive(Debug)]
+pub(crate) enum MaybePieceStoredResult {
```

Comment:
ProbabilisticStoredPieceResult ??

Reply:
I agree the name is a bit awkward. It is not necessarily probabilistic at the moment it is returned, but since there is no lock held throughout, things might change by the time this information is acted upon. Not sure `ProbabilisticStoredPieceResult` is much better option, but definitely looking for better alternatives.